R/data_read.R

```diff
@@ -158,6 +158,23 @@ data_read <- function(path,
 # process imported data from SPSS, SAS or Stata -----------------------
 
 .post_process_imported_data <- function(x, convert_factors, verbose) {
+  # for "unknown" data formats (like .RDS), which still can be imported via
+  # "rio::import()", we must check whether we actually have a data frame or
+  # not. Else, tell user.
+  if (!is.data.frame(x)) {
+    tmp <- tryCatch(as.data.frame(x, stringsAsFactors = FALSE), error = function(e) NULL)
```

Comment:
use `.coerce_to_dataframe()` instead

Reply:
I think it's a nice "side-effect" we should allow. I always forget whether functions are names "read" or "load", and had several tries until I could read an rds-file. So I thought `data_read()` is simpler to remember - but this didn't work for a model objects stored in the file. So I thought would be nice if `data_read()` can do this, too. But feel free to turn it into an error if you feel that _data_ read() should only read real data sets.

Reply:
No no I'm fine with `data_read` being able to read `rds`. 

What I meant is that you should put it in `.read_unknown()` and not `.post_process_imported_data()` because the latter is called by other `.read_` functions

Reply:
not sure my previous message clarifies but I'm fine with the code with your latest changes

Reply:
Yes, I understood moving the code part. My comment was referring to your suggestion using `.coerce_to_dataframe()`, which errors, while my code just warns

Reply:
I thought `tryCatch` would capture the error anyway

Reply:
Yeah, but then we would have a less specific error message:

> "`data` must be a data frame, or an object that can be coerced to a data frame."